### PR TITLE
Added new App Logo Component - Issue #2760

### PR DESCRIPTION
**Pull request by joanofdart**
(cherry picked from commit bf8340ebec155b38f695a7c9e38948d57da6ec8b)

Mega ultra amazing app-logo-component done ;) : issue #2760

### CODE REVIEW

File: libs/ui/src/lib/layout/app-logo/app-logo.component.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Component, OnInit } from '@angular/core';
+import { RouterQuery } from '@datorama/akita-ng-router-store';
+
+@Component({
+  selector: 'app-logo',
+  templateUrl: './app-logo.component.html',
+  styleUrls: ['./app-logo.component.scss']
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/layout/app-logo/app-logo.component.ts:7]: We set `changeDetection: ChangeDetectionStrategy.onPush` by default. It helps avoding performance leaks.

File: libs/ui/src/lib/layout/app-logo/app-logo.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  public imageLogo: string;
+
+  constructor(private routerQuery: RouterQuery) {}
+
+  ngOnInit() {
+    const appName = this.routerQuery.getData<string>('app');
+    this.imageLogo = this.mapAppToImage(appName);
+  }
+
+  mapAppToImage(appName: string) {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/layout/app-logo/app-logo.component.ts:19]: when the method doesn't dependant on the context of the class instance (`this`), prefer using a pure function than a method. In this case we can go a step further by creating a simple variable outside of the class scoped to the file : 
```typescript
const appLogos = {
  catalog: 'LogoArchipelContentPrimary.svg',
  festival: 'archipel_market.png'
};

@Component({ ... })
class AppLogoComponent  implements OnInit {
  public imageLogo: string;
  constructor(private routerQuery: RouterQuery) {}
  ngOnInit() {
    const appName = this.routerQuery.getData<string>('app');
    this.imageLogo = appLogos[appName];
  }
}
```

**Comment by fritzschoff** [comment]
I linked the PR to the issue, so it gets closed when you merge it.

**Comment by GrandSchtroumpf** [comment]
Good. I think we could have divide this PR into two. Once I approved the first one, the best is to merge as it (+ some last small changes if needed). It makes it easier to review the second PR.

File: apps/catalog/catalog/src/app/marketplace/right/view/view.module.ts [hunk 3]
```diff
@@ -2,25 +2,25 @@ import { NgModule } from '@angular/core';
...
 import { FlexLayoutModule } from '@angular/flex-layout';
-import { ImgAssetModule } from '@blockframes/ui/theme';
+import { AppLogoModule } from '@blockframes/ui/layout/app-logo/app-logo.module';
 import { ViewComponent } from './view.component';
 
-const routes = [{
-  path: '',
-  component: ViewComponent,
-  children: [{
+const routes = [
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/right/view/view.module.ts:8]: If you have an auto formatter on save please deactivate it.

File: libs/landing/src/lib/footer/footer.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <footer fxLayout="column" fxLayoutAlign="start center" fxLayout.lt-md="column" fxLayoutAlign.lt-md="center center">
   <a routerLink="/">
-    <img [asset]="logo" type="logo" class="ac-logo"/>
+    <app-logo></app-logo>
```
Comment (GrandSchtroumpf) [inline libs/landing/src/lib/footer/footer.component.html:3]: the class "ac-logo" is not used anymore, remove it in the css

File: libs/ui/src/lib/layout/app-logo/app-logo.component.scss [hunk 5]
```diff
@@ -0,0 +1,6 @@
+:host {
+  height: 40px;
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/layout/app-logo/app-logo.component.scss:2]: I checked on zeplin and the height is 48px. 
Also I'm surprised you had to remove the style only on one component stylesheet.
In contract-tunnel.component.scss for example you didn't remove the `img { height: 48px }`

File: libs/landing/src/lib/footer/footer.component.html (hunk 4 above)
Comment (joanofdart) [reply libs/landing/src/lib/footer/footer.component.html:3]: done ;)

**Pull request merged**